internal/driver/driver_focus.go

```diff
@@ -77,19 +77,44 @@ func compileTagFilter(name, value string, ui plugin.UI, err error) (func(*profil
 	if value == "" || err != nil {
 		return nil, err
 	}
+
+	tagValuePair := strings.SplitN(value, "=", 2)
+	var key string
+	if len(tagValuePair) == 2 {
+		key = tagValuePair[0]
+		value = tagValuePair[1]
+	} else {
+		key = ""
+	}
+
 	if numFilter := parseTagFilterRange(value); numFilter != nil {
 		ui.PrintErr(name, ":Interpreted '", value, "' as range, not regexp")
-		return func(s *profile.Sample) bool {
-			for key, vals := range s.NumLabel {
-				for _, val := range vals {
-					if numFilter(val, key) {
+		labelFilter := func(vals []int64, key string) bool {
```

Comment:
Ah, I missed that. Probably no then. This function got really long, but I can't think yet of a good way to make it shorter while keeping clear so probably leave it as is.